ui/components/src/Table/model/table-model.ts

```diff
@@ -112,17 +126,33 @@ export function getTableCellLayout(theme: Theme, density: TableDensity): TableCe
 
   // standard density
   const paddingY = theme.spacing(1);
-  const paddingX = theme.spacing(1.25);
+  const basePaddingX = theme.spacing(1.25);
+  const edgePaddingX = theme.spacing(2);
+  const paddingLeft = isFirstColumn ? edgePaddingX : basePaddingX;
+  const paddingRight = isLastColumn ? edgePaddingX : basePaddingX;
   const lineHeight = theme.typography.body1.lineHeight;
 
   return {
-    padding: `${paddingY} ${paddingX}`,
+    padding: `${paddingY} ${paddingRight} ${paddingY} ${paddingLeft}`,
     height: calculateTableCellHeight(lineHeight, paddingY),
     fontSize: theme.typography.body1.fontSize,
     lineHeight: lineHeight,
   };
 }
 
+export type TableCellAlignment = 'left' | 'right' | 'center';
+
+// Overrides of meta value, so it can have a meaningful type in our code instead
+// of `any`. Putting this in the model instead of a separate .d.ts file because
+// I couldn't get it to work properly that way and punted on figuring it out
+// after trying several things.
```

Comment:
There's something funky here I could not figure out after trying a bunch of different things, so I gave up and would love not to spend more time on it right now.